dali/test/python/test_pipeline_decorator.py

```diff
@@ -0,0 +1,109 @@
+# Copyright (c) 2021, NVIDIA CORPORATION. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from nvidia.dali import Pipeline, pipeline_def
+from nose.tools import nottest, raises
+import nvidia.dali.fn as fn
+from test_utils import get_dali_extra_path, compare_pipelines
+import os
+
+data_root = get_dali_extra_path()
+images_dir = os.path.join(data_root, 'db', 'single', 'jpeg')
+
+N_ITER = 7
+
+max_batch_size = 16
+num_threads = 4
+device_id = 0
+
+
+def reference_pipeline(flip_vertical, flip_horizontal, ref_batch_size=max_batch_size):
+    pipeline = Pipeline(ref_batch_size, num_threads, device_id)
+    with pipeline:
+        data, _ = fn.file_reader(file_root=images_dir)
+        img = fn.image_decoder(data, device="mixed")
+        flipped = fn.flip(img, horizontal=flip_horizontal, vertical=flip_vertical)
+        pipeline.set_outputs(flipped, img)
+    return pipeline
+
+
+@nottest
+@pipeline_def(batch_size=max_batch_size, num_threads=num_threads, device_id=device_id)
+def pipeline_static(flip_vertical, flip_horizontal):
+    data, _ = fn.file_reader(file_root=images_dir)
+    img = fn.image_decoder(data, device="mixed")
+    flipped = fn.flip(img, horizontal=flip_horizontal, vertical=flip_vertical)
+    return flipped, img
+
+
+@nottest
+@pipeline_def
+def pipeline_runtime(flip_vertical, flip_horizontal):
+    data, _ = fn.file_reader(file_root=images_dir)
+    img = fn.image_decoder(data, device="mixed")
+    flipped = fn.flip(img, horizontal=flip_horizontal, vertical=flip_vertical)
+    return flipped, img
+
+
+
+
+
+@nottest
```

Comment:
in other test suites we use `check_`. You wouldn't have to mark it as nottest if you follow this suggestion

Reply:
`check_` is vague. When you see it, it doesn't answer a question, why is there a function called in the same way, just `check_` instead of `test_`? And `@nottest` gives you a clear answer ;)